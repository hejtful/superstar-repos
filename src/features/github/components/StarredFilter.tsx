import { ChangeEvent, EventHandler } from 'react';

import styles from './StarredFilter.module.css';

interface Props {
  value: boolean;
  onChange: EventHandler<ChangeEvent>;
}

export function StarredFilter({ value, onChange }: Props) {
  return (
    <label className={styles.switch} data-testid="starred-filter">
      <span className={styles.labelText}>Starred:</span>
      <input
        type="checkbox"
        className={styles.input}
        checked={value}
        onChange={onChange}
      />
      <span className={styles.slider}></span>
    </label>
  );
}
