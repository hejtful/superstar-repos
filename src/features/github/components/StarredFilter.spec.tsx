import { render } from '@testing-library/react';

import { StarredFilter } from './StarredFilter';

it('should render properly', () => {
  const { asFragment } = render(
    <StarredFilter value={false} onChange={jest.fn()} />
  );

  expect(asFragment()).toMatchSnapshot();
});
