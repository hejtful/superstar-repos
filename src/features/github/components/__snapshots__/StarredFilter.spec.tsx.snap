// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly 1`] = `
<DocumentFragment>
  <label
    class="switch"
    data-testid="starred-filter"
  >
    <span
      class="labelText"
    >
      Starred:
    </span>
    <input
      class="input"
      type="checkbox"
    />
    <span
      class="slider"
    />
  </label>
</DocumentFragment>
`;
